src/masternodes/vaulthistory.h

```diff
@@ -55,14 +55,62 @@ struct VaultHistoryValue {
     }
 };
 
+struct VaultStateKey {
+    uint256 vaultID;
+    uint32_t blockHeight;
+
+    ADD_SERIALIZE_METHODS;
+
+    template <typename Stream, typename Operation>
+    inline void SerializationOp(Stream& s, Operation ser_action) {
+        READWRITE(vaultID);
+
+        if (ser_action.ForRead()) {
+            READWRITE(WrapBigEndian(blockHeight));
+            blockHeight = ~blockHeight;
+        }
+        else {
+            uint32_t blockHeight_ = ~blockHeight;
+            READWRITE(WrapBigEndian(blockHeight_));
+        }
+    }
+};
+
+struct VaultStateValue {
+    bool isUnderLiquidation;
+    TAmounts collaterals;
+    CCollateralLoans collateralsValues;
+    std::vector<CAuctionBatch> auctionBatches;
+    uint32_t ratio;
+    std::string schemeID;
```

Comment:
Probably it's better to keep entire scheme, the name could be changed or scheme could be deleted. Then the name will not provide any significant info.

Reply:
I did mention this, my original plan was to store the ratio of the scheme at the time but it was decided that the scheme name would suffice.

Reply:
I don't think there a `promise` that name will be sufficient, `CLoanScheme` look perfect fit here.

Reply:
I agree, CLoanScheme would be better here. Will get back on this one.